<h1>A01639631 Alan Paul García Rosales

# Activity: work with the iris dataset

1. Load the iris.csv file in your computer and compare with the data printed above.


2. How many observations (rows) are in total? 
    

3. How many variables (columns) are in total? What do they represent?
    

4. How many observations are for each type of flower? 


5. What is the type of data for each variable?


6. What are the units of each variable?


<H3>1. Load the iris.csv file in your computer and compare with the data printed above.

In [4]:
# Import the packages that we will be using
import matplotlib.pyplot as plt
import pandas as pd                 # For data handling

# Dataset url
url = "datasets/iris/iris.csv"

# Load the dataset
dt = pd.read_csv(url )

In [5]:
dt

sepal length  sepal width  petal length  petal width            type
0             5.1          3.5           1.4          0.2     Iris-setosa
1             4.9          3.0           1.4          0.2     Iris-setosa
2             4.7          3.2           1.3          0.2     Iris-setosa
3             4.6          3.1           1.5          0.2     Iris-setosa
4             5.0          3.6           1.4          0.2     Iris-setosa
..            ...          ...           ...          ...             ...
145           6.7          3.0           5.2          2.3  Iris-virginica
146           6.3          2.5           5.0          1.9  Iris-virginica
147           6.5          3.0           5.2          2.0  Iris-virginica
148           6.2          3.4           5.4          2.3  Iris-virginica
149           5.9          3.0           5.1          1.8  Iris-virginica

[150 rows x 5 columns]

<h3> 2. How many observations (rows) are in total? 

There are 150 rows.

<h3>3. How many variables (columns) are in total? What do they represent?

There are 5 columns, each column represents:
   1. sepal length in cm
   2. sepal width in cm
   3. petal length in cm
   4. petal width in cm
   5. class: _Iris Setosa, Iris Versicolour, Iris Virginica_

<h3> 4. How many observations are for each type of flower?

In [17]:
print(dt.groupby(['type']).size())

type
Iris-setosa        50
Iris-versicolor    50
Iris-virginica     50
dtype: int64


<h3>5. What is the type of data for each variable?

The first 4 columns are measurable, so sepal length, sepal width, petal length and petal width are cuantitative data. On the other hand, type is cualitative, because it refers to the class of flower that was measured.

In [19]:
dt.dtypes

sepal length    float64
sepal width     float64
petal length    float64
petal width     float64
type             object
dtype: object

<h3>6. What are the units of each variable?

Sepal length and width, centimeters. Petal length and width, centimeters.